### PR TITLE
code: adding static obstacles

### PR DESCRIPTION
**Pull request by laurafcamargos**

### CODE REVIEW

**Review by roselino-quatro — CHANGES_REQUESTED** [review]
- Sometimes the apple spawns inside one of the obstacles, softlocking the game
- Currently the obstacles can spawn inside each other (its easier to test this by increasing the amount of obstacles temporarily)
- the snake spawn point can be inside a obstacle, making the game impossible to play
- it would make sense creating a option to start a game without obstacles, but maybe its more fitting to do it in a separate issue

File: coral.py [hunk 1]
```diff
@@ -90,9 +90,26 @@ def center_prompt(title, subtitle):
...
+    def __init__(self):
+        # Generate a random position for the obstacle within the game arena.
+        self.x = int(random.randint(0, WIDTH - GRID_SIZE) / GRID_SIZE) * GRID_SIZE
+        self.y = int(random.randint(0, HEIGHT - GRID_SIZE) / GRID_SIZE) * GRID_SIZE
+        # Create a rectangle object representing the obstacle.
+        self.rect = pygame.Rect(self.x, self.y, GRID_SIZE, GRID_SIZE)
+
+    def update(self):
+        # Draw the obstacle on the arena.
+        pygame.draw.rect(arena, "#FF4500", self.rect)  # Use orange color for obstacles.
```
Comment (roselino-quatro) [inline coral.py:106]: perhaps creating a constant for the obstacle color would fit better in the current code style, like we are already doing with the other objects colors

**Review by laurafcamargos — COMMENTED** [review]
The comments made and the suggested change are in agreement.

**Review by laurafcamargos — COMMENTED** [review]
Added logic to check if the snake's head collides with an obstacle, added drawing of obstacles in the main game loop, initial generation of obstacles, and created a function to generate a list of obstacles based on the specified number. This enables the generation of multiple obstacles either at the start of the game or during its execution.

**Pull request closed without merging**